core/src/main/java/io/paradoxical/cassieq/dataAccess/QueueRepositoryImpl.java

```diff
@@ -29,57 +39,162 @@ public QueueRepositoryImpl(@NonNull final Session session) {
         this.session = session;
     }
 
+
     @Override
-    public void createQueue(@NonNull final QueueDefinition definition) {
+    public void markForDeletion(final QueueDefinition definition) {
+        if(trySetQueueDefinitionStatus(definition.getId(), QueueStatus.Deleting)) {
+            if (trySetQueueNameAvailableStatus(definition, QueueStatus.Inactive)) {
+                logger.with("definition", definition).success("Marked for deletion");
+            }
+        }
+    }
 
-        initializeMonotonicValue(definition.getQueueName());
+    private boolean trySetQueueNameAvailableStatus(final QueueDefinition definition, final QueueStatus status) {
+        final Statement update =
+                QueryBuilder.update(Tables.QueueNameManager.TABLE_NAME)
+                            .where(eq(Tables.QueueNameManager.QUEUE_NAME, definition.getQueueName().get()))
+                            .with(set(Tables.QueueNameManager.STATUS, status.ordinal()))
+                            .onlyIf(eq(Tables.QueueNameManager.VERSION, definition.getVersion()));
 
-        initializePointers(definition.getQueueName());
+        return session.execute(update).wasApplied();
+    }
 
+    @Override
+    public void createQueue(@NonNull final QueueDefinition definition) throws QueueExistsError {
         insertQueueRecord(definition);
```

Comment:
Also realizing that we can't do this first since it will mean the queue is available for use but the pointers aren't ready. Maybe I can fix this by using the queue status to create it but it not actice, initialize repos, then set active